test/pylib/harness_cli.py

```diff
@@ -0,0 +1,168 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""Harness client.
+   Communicates with Harness server via socket.
+   Provides helper methods to test cases.
+   Manages driver refresh when cluster is cycled.
+"""
+
+from typing import List, Optional, Callable
+import aiohttp                                                             # type: ignore
+import aiohttp.web                                                         # type: ignore
+from cassandra.cluster import Session as CassandraSession                  # type: ignore
+from cassandra.cluster import Cluster as CassandraCluster                  # type: ignore
+
+
+class HarnessOpException(Exception):
+    """Operational exception with remote API"""
+
+class HarnessCli():
+    """Helper Harness API client
+    Args:
+        sock_path (str): path to an AF_UNIX socket where Harness server is listening
+        con_gen (Callable): generator function for CQL driver connection to a cluster
+    """
+    conn: aiohttp.UnixConnector
+    session: aiohttp.ClientSession
+    dirty: bool
+
+    def __init__(self, sock_path: str,
+                 con_gen: Optional[Callable[[List[str], int], CassandraSession]] = None) -> None:
+        self.con_gen = con_gen
+        self.ccluster: Optional[CassandraCluster] = None
+        self.cql: Optional[CassandraSession] = None
+        # API
+        self.sock_path = sock_path
+
+    async def start(self):
+        """Setup connection to Harness server"""
+        self.conn = aiohttp.UnixConnector(path=self.sock_path)
+        self.session = aiohttp.ClientSession(connector=self.conn)
+
+    async def driver_connect(self) -> None:
+        """Connect to cluster"""
+        if self.con_gen is not None:
+            self.ccluster = self.con_gen(await self.servers(), 9042)
+            self.cql = self.ccluster.connect()
+
+    def driver_close(self) -> None:
+        """Disconnect from cluster"""
+        if self.ccluster is not None:
+            self.ccluster.shutdown()
+            self.ccluster = None
+        self.cql = None
+
+    # Make driver update endpoints from remote connection
+    def _driver_update(self) -> None:
+        if self.ccluster is not None:
+            self.ccluster.control_connection.refresh_node_list_and_token_map()
+
+    async def before_test(self, test_name: str) -> None:
+        """Before a test starts check if cluster needs cycling and update driver connection"""
+        dirty = await self.is_dirty()
```

Comment:
I don't quite understand the timing of this. This is a check that runs before test. And it runs an rpc against harness which disposes of a dirty cluster *after* the test. So if the cluster was dirty and was disposed of, this rpc apparently woudl return false, which is wrong. Isn't it safer to use some kind of cluster uuid, and check if it has changed for whatever reason?

Reply:
In fact we already have the cluster uuid...

Reply:
Cluster is disposed in before test. Disposing in after test doesn't make sense, else if last test makes the cluster dirty, manager will bring up a new cluster for nothing (no next test).